src/mysky/index.ts

```diff
@@ -189,11 +189,15 @@ export class MySky {
 
         // Send the UI the list of required permissions.
 
+        // Slightly hacky, but this copies permissions array so that Proxy objects
```

Comment:
This comment leaves me very confused, can you expand it?

Reply:
It's a single framework, but among JS devs [its usage is at 49%](https://2020.stateofjs.com/en-US/technologies/front-end-frameworks/), behind only React and Angular.

I'll try to find Vuex communication channels, but I imagine from their perspective, our use case is more niche than us supporting them. We could also maybe do a type-check here so the flow and reasoning is more clear.

Reply:
I am fine with including this to make Vue work, but since it is an ugly hack the comment should explain it better. You'll also need to resolve the merge conflict